### PR TITLE
Default config first

### PR DESCRIPTION
**Pull request by lhoestq**
Show the default config first in the Viewer.

I did this by ordering the configs in `dataset-config-names` to have the default config in first position

fix https://github.com/huggingface/datasets-server/issues/2541

No need to recompute all the entries, it's fine to apply this to new / updated datasets imo

cc @HugoLaurencon for WebSight

### CODE REVIEW

File: services/worker/tests/job_runners/dataset/test_config_names.py [hunk 1]
```diff
@@ -77,12 +77,13 @@ def test_compute_too_many_configs(
     )
 
     with patch("worker.job_runners.dataset.config_names.get_dataset_config_names", return_value=configs):
-        if error_code:
-            with pytest.raises(CustomError) as exc_info:
-                job_runner.compute()
-            assert exc_info.value.code == error_code
-        else:
-            assert job_runner.compute() is not None
+        with patch("worker.job_runners.dataset.config_names.get_dataset_default_config_name", return_value=None):
```
Comment (severo) [inline services/worker/tests/job_runners/dataset/test_config_names.py:80]: Nice feature.

Maybe add a unit test here to be sure we set the default config first.

**Comment by polinaeterna** [comment]
i love this but this also would break the [bigcode/the-stack-v2-dedup](https://huggingface.co/datasets/bigcode/the-stack-v2-dedup) viewer which we struggled for so long to make it work :crying_cat_face: 
the problem is - default config in bigcode/the-stack-v2-dedup is the one that includes all the data and it doesn't work, see https://github.com/huggingface/datasets-server/issues/2552. so @anton-l and I decided to split all the data into many small configs (one per language) so that the first small config is shown in the viewer. and it works now! 

so please don't deploy until we find a solution :pray:

**Comment by polinaeterna** [comment]
this is why i was dreaming about having a dedicated field for the viewer :D https://github.com/huggingface/datasets-server/issues/1953#issuecomment-1983496422

File: services/worker/src/worker/job_runners/dataset/config_names.py [hunk 2]
```diff
@@ -4,7 +4,7 @@
 import logging
 from typing import Optional
 
-from datasets import get_dataset_config_names
+from datasets import get_dataset_config_names, get_dataset_default_config_name
```
Comment (severo) [inline services/worker/src/worker/job_runners/dataset/config_names.py:7]: it's not clear from https://github.com/huggingface/datasets/blob/f45bc6caa25115a04c41b278671a5a89457eb66c/src/datasets/inspect.py#L371 when the result i None. 

I was wondering when we should reorder, and when not to do it.

I imagine that we should do it only when `default: true` is present in the YAML file, no? Is it the case?

Comment (lhoestq) [reply services/worker/src/worker/job_runners/dataset/config_names.py:7]: It's None when there are multiple configs and no default config.

A default config is a config named "default" or a config with `default: true`

Comment (severo) [reply services/worker/src/worker/job_runners/dataset/config_names.py:7]: OK thanks! Opened: https://github.com/huggingface/datasets/pull/6722

**Review by severo — APPROVED** [review]

**Pull request merged**